crates/holochain_p2p/src/types/actor.rs

```diff
@@ -252,153 +201,156 @@ impl Default for GetActivityOptions {
-        ) -> Vec<MetadataSet>;
-
-        /// Get links from the DHT.
-        fn get_links(
-            dna_hash: DnaHash,
-            link_key: WireLinkKey,
-            options: GetLinksOptions,
-        ) -> Vec<WireLinkOps>;
-
-        /// Get a count of links from the DHT.
-        fn count_links(
-            dna_hash: DnaHash,
-            query: WireLinkQuery,
-        ) -> CountLinksResponse;
-
-        /// Get agent activity from the DHT.
-        fn get_agent_activity(
-            dna_hash: DnaHash,
-            agent: AgentPubKey,
-            query: ChainQueryFilter,
-            options: GetActivityOptions,
-        ) -> Vec<AgentActivityResponse>;
-
-        /// A remote node is requesting agent activity from us.
-        fn must_get_agent_activity(
-            dna_hash: DnaHash,
-            author: AgentPubKey,
-            filter: holochain_zome_types::chain::ChainFilter,
-        ) -> Vec<MustGetAgentActivityResponse>;
-
-        /// Send a validation receipt to a remote node.
-        fn send_validation_receipts(dna_hash: DnaHash, to_agent: AgentPubKey, receipts: ValidationReceiptBundle) -> ();
-
-        /// New data has been integrated and is ready for gossiping.
-        fn new_integrated_data(dna_hash: DnaHash) -> ();
-
-        /// Check if any local agent in this space is an authority for a hash.
-        fn authority_for_hash(dna_hash: DnaHash, basis: OpBasis) -> bool;
-
-        /// Messages between agents negotiation a countersigning session.
-        fn countersigning_session_negotiation(
-            dna_hash: DnaHash,
-            agents: Vec<AgentPubKey>,
-            message: event::CountersigningSessionNegotiationMessage,
-        ) -> ();
-
-        /// Dump network metrics.
-        fn dump_network_metrics(
-            dna_hash: MaybeDnaHash,
-        ) -> String;
-
-        /// Dump network stats.
-        fn dump_network_stats() -> String;
-
-        /// Get struct for diagnostic data
-        fn get_diagnostics(dna_hash: DnaHash) -> KitsuneDiagnostics;
-
-        /// Get the storage arcs of the agents currently in this space.
-        fn storage_arcs(dna_hash: DnaHash) -> Vec<kitsune2_api::DhtArc>;
-    }
-}
-
-/// Convenience type for referring to the HolochainP2p GhostSender
-pub type HolochainP2pRef = ghost_actor::GhostSender<HolochainP2p>;
-
-/// Extension trait for converting `GhostSender<HolochainP2p>` into HolochainP2pDna
-pub trait HolochainP2pRefToDna {
-    /// Partially apply dna_hash && agent_pub_key to this sender,
-    /// binding it to a specific dna context.
-    fn into_dna(self, dna_hash: DnaHash, chc: Option<ChcImpl>) -> crate::HolochainP2pDna;
-
-    /// Clone and partially apply dna_hash && agent_pub_key to this sender,
-    /// binding it to a specific dna context.
-    fn to_dna(&self, dna_hash: DnaHash, chc: Option<ChcImpl>) -> crate::HolochainP2pDna;
+/// Trait defining the main holochain_p2p interface.
+pub trait HcP2p: 'static + Send + Sync + std::fmt::Debug {
```

Comment:
This file used to be the ghost_actor actor channel for holochain to make requests into legacy kitsune.

Now it is a simple kitsune2-style trait that can be type-erased into a dynamic trait-object.